### ISSUE TITLE
The 32x version could be the authentic Cave Story

### ISSUE THREAD

**Issue by BencoXX** [andwn/cave-story-md#396]
**Have you considered creating a Sega 32x and Neptune version, or just starting a project?**
The Mega Drive challenge is a great achievement, but it’s time for a 100% conversion that matches the original, especially for homebrew console.
I would also like to think of the new Neptune owners and myself.

Help to get started based on another similar project.
https://github.com/jvisser/golden-axe-32x-edition

Thank you and stay strong.

**Comment by mastercheff** [comment]
it's close to 100% Imo

**Comment by BencoXX** [comment]
It definitely works well, but the colors are different.

**Comment by thomasgoldstein** [comment]
I have to say that sounds really entitled. Anyone else can start that project if they want to. This issue is just rude and off topic here.

**Comment by mastercheff** [comment]
Hard disagree on the music too. YM2612 and it's utilization is amazing here

**Comment by BencoXX** [comment]
This is not off-topic, in my opinion, because the MD version is approaching 1.0, and we can discuss the next steps and the issues that have arisen since then. This issue does not call into question the current size of the code or the dedication of the programmer.

We have a list like this.
https://github.com/andwn/cave-story-md/blob/master/doc/DIFFERENCES.md
The issues in the Colors and Sound sections could be summarized with **hardware upgrade**, as well as any other performance-related issues.

**Comment by mastercheff** [comment]
But is the author interested in doing this, I think he would rather port this to the X6800 computer just for the sake of doing so. Literally anyone can pick it up and fork it (like most recent GBA port)

**Comment by BencoXX** [comment]
[Cave Story GBA](https://github.com/tilderain/cave-story-gba)
OMG, this is really nice work! Yes, this is what I had in mind for the 32x case.

**Comment by VGkav** [comment]
https://www.doomworld.com/forum/topic/148783-doom-cd32x-fusion/

There is this, too, for those who are interested :D

**Comment by mastercheff** [comment]
It's all off topic, for all those interested in actually improving the game (32X won't be gamechanger) maybe actually play the most recent WIP build of the game and hunt for bugs ?

**Comment by BencoXX** [comment]
OK ... Off topic I apologize ...
 
I'll outline the situation.
I really love Cave Story; I think it's a very nice game.

But the Android version doesn't support controllers, and I don't want to play on the touchscreen. The PC version, Cave Story+, on Steam is really good; I bought it, it's great, but it has crashed on me several times, and I can't customize the controller. Oh, and it supports Linux, but it crashes there too :D.

Well, I downloaded RetroArch, and it has Cave Story on it, yeah, it's really great :D, but only the MD version. Which is very good; I've completed it several times. Unfortunately, colors and sounds are different.
I haven't tried the GBA version yet, but I'll give it a shot, although it's not available on RetroArch.

In the meantime, I'm considering building a Retro Homebrew Neptune console. It would be great if the Cave Story 32X version could be ready by then. :D Or it could be a WIP version, but I guess that's only for Windows... It's really cool that there are a million retro consoles and emulator consoles, but they won't run it...

**Comment by mastercheff** [comment]
I can understand Your optimism and excitement regarding potential X32 port, but the truth is, Cave Story MD in the form we got (for free) is a gargantuan task that Andy achieved through Years of hard work and precise coding / cutting corners / adding visual effects to fit on the specific system. I think You should steer towards the PSX version that surfaced some time ago that one seems be more in line of what You seek (original PC version translated without cuts or changes)

**Comment by JoshuaPettus** [comment]
When the author started this project, of couse he knew about the 32x.  Its no secret.  That clearly wasnt ever a goal for the project. So to suggest and insist that it be so is just absurd.

**Comment by thomasgoldstein** [comment]
@BencoXX: Basically, you are posting here saying "This project is not what I want. I want something else.", which is wrong for two reasons:
* This is not about fixing or improving the MD port of Cave Story, therefore the repo cave-story-**_md_** is not the right place for this. Hence "off topic"
* Even if you got in touch with Andy through some other way, you are basically asking him to devote months or likely years of his life for something _you_ want. For free. Hence "rude and entitled".

Besides, there are billions of us on Earth. Why should it be up to Andy to handle that 32X port? Just because he has worked on something similar? This is really a thankless hobby. As Joshua pointed out, I'm pretty sure Andy knows that the 32X exists, and he's big enough to decide what his next projects should be. Let the man rest, and maybe consider handling that 32X port on your own, since you care deeply about it. After all, Andy's port is open source, so you can fork it and build on top of it if you want to.

**Comment by BencoXX** [comment]
@thomasgoldstein You are absolutely right; since then, I have also realized that this is not such a small task (let's say, redrawing the palette). And indeed, it couldn't have been the developer's goal to create a 32x version.
I still maintain that the MD port could be the basis for the 32x version.
You were right... But not in this project, perhaps in another fork.

@mastercheff I downloaded now the PSX version; it's an excellent port and works well in the DuckStation emulator. If I load the ISO into RAM and increase the loading speed, it doesn't take long to load at all. :D Thank you!

I wouldn't play this on a physical PS1. Not just because of CDs, there aren't many good 2D games. :(

Unfortunately, I'm always searching for the perfect retro console. I think the Neptune could be good because it's compatible with MD, and there's no hassle with CDs; it has some really great 16-bit era games.

One problem is that the games don't automatically update to 32X; unfortunately, this would require the help of programmers. There are many games that could look better with higher colors and even additional effects programmed in.

The best example is MK2, which has both MD and 32X versions. Since then, we know that both were built on MD code, just with higher color and sound palettes.
I dream of such updated MD games. The 32X version of Golden Axe is a light in the night. :) 

_(Someone can close this issue.)_

**Comment by mastercheff** [comment]
UGH We may all be wrong here , seems like the man does not take the bait but instead 
He already works on Hitachi SH-2 architecture (which is the brains of the X32 ?)

![image](https://github.com/user-attachments/assets/966a20d1-de28-4d4c-964f-dc49f945791c)

If this was a questionnaire "Do we want the X32 port" I'd say yes in a heartbeat
